### PR TITLE
Us015 edit a role

### PR DESCRIPTION
**Pull request by Adaray786**

### CODE REVIEW

File: src/test/java/org/kainos/ea/db/JobRoleDaoTest.java [hunk 1]
```diff
@@ -74,4 +75,48 @@ void getJobRoles_ShouldThrowSQLException_WhenSQLExceptionOccurs() throws SQLExce
...
+
+        DatabaseConnector.setConn(connection);
+
+        PreparedStatement preparedStatement = Mockito.mock(PreparedStatement.class);
+        Mockito.when(connection.prepareStatement(Mockito.anyString())).thenReturn(preparedStatement);
+        Mockito.doThrow(new SQLException()).when(preparedStatement).executeUpdate();
+
+        assertThrows(SQLException.class, () -> jobRoleDao.editJobRole(id, jobRoleEditRequest));
+    }
 }
```
Comment (TomBillingtonUK) [inline src/test/java/org/kainos/ea/db/JobRoleDaoTest.java:122]: Missing tests for get job role by id

File: src/test/java/org/kainos/ea/resources/JobRoleControllerTest.java [hunk 2]
```diff
@@ -52,4 +55,40 @@ void viewRoles_ShouldReturnResponse500_WhenServiceThrowsFailedToGetJobRoles()
...
+        int id = 5;
+        JobRoleEditRequest jobRole = new JobRoleEditRequest("string", "string", "string", "string", 1, 1);
+
+        Mockito.when(jobRoleServiceMock.editJobRole(id, jobRole)).thenReturn(5);
+
+        Response response = jobRoleController.editJobRole(id, jobRole);
+
+        Assertions.assertEquals(200, response.getStatus());
+    }
+
```
Comment (TomBillingtonUK) [inline src/test/java/org/kainos/ea/resources/JobRoleControllerTest.java:70]: Missing test for 400 being returned

File: src/test/java/org/kainos/ea/db/JobRoleDaoTest.java (hunk 1 above)
Comment (Adaray786) [reply src/test/java/org/kainos/ea/db/JobRoleDaoTest.java:122]: Completed tests

File: src/test/java/org/kainos/ea/resources/JobRoleControllerTest.java (hunk 2 above)
Comment (Adaray786) [reply src/test/java/org/kainos/ea/resources/JobRoleControllerTest.java:70]: Relabelled one of my 500 tests wrong so changed that to 400

File: src/main/java/org/kainos/ea/api/JobRoleService.java [hunk 3]
```diff
@@ -30,4 +33,20 @@ public List<JobRoleRequest> viewRoles() throws JobRolesNotFoundException, Failed
...
     }
+
+    public int editJobRole(int id, JobRoleEditRequest jobRoleEditRequest) throws JobRoleDoesNotExistException, FailedToGetJobRole {
+        try {
+            JobRole jobRoleToUpdate = jobRoleDao.getJobRoleById(id);
+
+            if(jobRoleToUpdate == null) {
+                throw new JobRoleDoesNotExistException();
+            }
+            jobRoleDao.editJobRole(id, jobRoleEditRequest);
```
[the remark this answers is not in the document]
Comment (Adaray786) [reply src/main/java/org/kainos/ea/api/JobRoleService.java:44]: Added Validation, need to add tests for it though

File: pom.xml [hunk 4]
```diff
@@ -191,6 +191,10 @@
             <plugin>
                 <artifactId>maven-surefire-plugin</artifactId>
                 <version>2.22.2</version>
+                <configuration>
+                    <reuseForks>false</reuseForks>
+                    <forkCount>1</forkCount>
+                </configuration>
```
[the remark this answers is not in the document]
Comment (Adaray786) [reply pom.xml:197]: yes, thats the change that fixed the ordering tests causing the failure of one of them

**Review by sammyperry101 — APPROVED** [review]

File: src/main/java/org/kainos/ea/validator/JobRoleValidator.java [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+    private DatabaseConnector databaseConnector;
+    BandDao bandDao = new BandDao(databaseConnector);
+    FamilyDao familyDao = new FamilyDao(databaseConnector);
+
+
+    public boolean validateJobRole(JobRoleEditRequest jobRoleEditRequest) throws SQLException, InvalidNameException,
+            InvalidSharepointLinkException, InvalidBandIDException, InvalidFamilyIDException, NullfieldException {
+        if (jobRoleEditRequest == null) {
+            throw new NullfieldException();
```
Comment (TomBillingtonUK) [inline src/main/java/org/kainos/ea/validator/JobRoleValidator.java:25]: SHould be NullFieldException

**Review by TomBillingtonUK — APPROVED** [review]

**Review by TomBillingtonUK — APPROVED** [review]

**Pull request merged**